File: front/src/netvision/models/PacketData.ts
import { LatLng } from "./LatLng";

export type PacketData = {
  from: LatLng;
  srcip: string;
  srcport: string;
  protocolType: string;
  abuseIPScore: number;
  packetCount: number;
  isSpamhausContain: boolean;
  isBlocklistDeContain: boolean;
};

File: back/domain/model/packet_data.go
package model

import (
	"net/netip"
)

type PacketData struct {
	From domain.Location
	Srcip netip.Addr
	Srcport int
	ProtocolType string
	AbuseIPScore int
	PacketCount int
	IsSpamhausContain bool
	IsBlocklistDeContain bool
}

// func IsValidPacketData(packetData *PacketData) bool {
// 	if packetData.Srcip == "" {
// 		return false
// 	}
// 	if packetData.From.Latitude == 0 && packetData.From.Longitude == 0 {
// 		return false
// 	}

// 	return true
// }